### PR TITLE
Richardmillson/fix/specify cohere model

### PR DESCRIPTION
**Pull request by richardmillson**

### CODE REVIEW

**Review by hamelin — COMMENTED** [review]
This PR does way more than merely fix the Cohere wrapper. Are you sure all the changes involving the Anthropic and OpenAI LLM wrappers are ready to be merged to the main branch?

File: toponymy/topic_naming.py [hunk 1]
```diff
@@ -420,10 +419,10 @@ def __init__(
         self.documents = documents
         self.document_vectors = document_vectors
         self.document_map = document_map
-        if (cluster_layers is not None) and (type(cluster_layers).__name__
-                                             != "ClusterLayers"):
+        if (cluster_layers
+                is not None) and not isinstance(cluster_layers, ClusterLayers):
             raise ValueError(
-                f"cluster_layers must be of type ClusterLayers class not {type(cluster_layers).__name__}"
+                f"cluster_layers must be of type ClusterLayers not {type(cluster_layers)}"
```
Comment (hamelin) [inline toponymy/topic_naming.py:425]: Why remove the `.__name__`? It avoids the awkward `<type BLAH>` string representation of most classes.

**Review by hamelin — COMMENTED** [review]
I think this is good code. As we discussed elsewhere, let's try and keep abreast of code tidying separately from contributive PRs.

**Comment by lmcinnes** [comment]
For reference this broke using a local tokenizer which was infinitely preferable since cohere's tokenizer is pretty expensive for the purposes we are using it for. I solved this by removing all the tokenizer handling from toponymy in general.

**Comment by lmcinnes** [comment]
I think the real answer is that you should never use API tokenizers for this (far too expensive), and locally the tokenizer uses a reasonable amount of compute for limited gain. Better to just try for a large context and hope for the best if you don't meet it.

**Pull request merged**